src/app/Mappings/components/AddEditMappingModal.tsx

```diff
@@ -113,27 +131,37 @@ const AddEditMappingModal: React.FunctionComponent<IAddEditMappingModalProps> =
           <>
             <Grid className={spacing.mbMd}>
               <GridItem sm={12} md={5} className={spacing.mbMd}>
-                <FormGroup label="Name" isRequired fieldId="mapping-name">
-                  <TextInput
-                    id="mapping-name"
-                    value={mappingName}
-                    type="text"
-                    onChange={setMappingName}
+                <Form>
+                  <ValidatedTextInput
+                    field={forms.mapping.fields.name}
+                    label="Name"
+                    isRequired
+                    fieldId="mapping-name"
                   />
-                </FormGroup>
+                </Form>
               </GridItem>
               <GridItem />
               <GridItem sm={12} md={5}>
-                <FormGroup label="Source provider" isRequired fieldId="source-provider">
+                <FormGroup
+                  label="Source provider"
+                  isRequired
+                  fieldId="source-provider"
+                  helperTextInvalid={forms.mapping.fields.sourceProvider.error}
```

Comment:
We can use the `getFormGroupProps` helper from the useFormState file for these validation-related props.
```ts
{...getFormGroupProps(form.fields.sourceProvider)}
```

Reply:
Also we should use it in the FormGroup around the target provider field below as well.